WordPress/Sniffs/Theme/DeprecatedArgsSniff.php

```diff
@@ -0,0 +1,359 @@
+			),
+			'template_directory'   => array(
+				'alt' => 'get_template_directory_uri()',
+			),
+			'template_url'         => array(
+				'alt' => 'get_template_directory_uri()',
+			),
+			'text_direction'       => array(
+				'alt' => 'is_rtl()',
+			),
+			'feed_url'             => array(
+				'alt' => "get_feed_link( 'feed' ), where feed is rss, rss2 or atom",
+			),
+		),
+		'bloginfo' => array(
+			'home'                 => array(
+				'alt' => 'echo esc_url( home_url() )',
+			),
+			'url'                  => array(
+				'alt' => 'echo esc_url( home_url() )',
+			),
+			'wpurl'                => array(
+				'alt' => 'echo esc_url( site_url() )',
+			),
+			'stylesheet_directory' => array(
+				'alt' => 'echo esc_url( get_stylesheet_directory_uri() )',
+			),
+			'template_directory'   => array(
+				'alt' => 'echo esc_url( get_template_directory_uri() )',
+			),
+			'template_url'         => array(
+				'alt' => 'echo esc_url( get_template_directory_uri() )',
+			),
+			'text_direction'       => array(
+				'alt' => 'is_rtl()',
+			),
+			'feed_url'             => array(
+				'alt' => "echo esc_url( get_feed_link( 'feed' ) ), where feed is rss, rss2 or atom",
+			),
+		),
+		'get_option' => array(
+			'home'     => array(
+				'alt' => 'home_url()',
+			),
+			'site_url' => array(
+				'alt' => 'site_url()',
+			),
+		),
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_EVAL,
+			T_CONSTANT_ENCAPSED_STRING,
+		);
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		// Exclude function definitions, class methods, and namespaced calls.
+		if (
+			T_STRING === $token['code']
+			&&
+			( $prev = $phpcsFile->findPrevious( T_WHITESPACE, ( $stackPtr - 1 ), null, true ) )
+			&&
+			(
+				// Skip sniffing if calling a method, or on function definitions.
+				in_array( $tokens[ $prev ]['code'], array( T_FUNCTION, T_DOUBLE_COLON, T_OBJECT_OPERATOR ), true )
+				||
+				(
+					// Skip namespaced functions, ie: \foo\bar() not \bar().
+					T_NS_SEPARATOR === $tokens[ $prev ]['code']
+					&&
+					( $pprev = $phpcsFile->findPrevious( T_WHITESPACE, ( $prev - 1 ), null, true ) )
+					&&
+					T_STRING === $tokens[ $pprev ]['code']
+				)
+			)
+			) {
+			return;
+		}
+
+		if ( isset( $this->functions[ $token['content'] ] ) ) {
+			$function      = $this->functions[ $token['content'] ];
+			$parameter_arg = $this->getFunctionCallParameter( $phpcsFile, $stackPtr, 1 );
```

Comment:
Quick remark: I would make the parameter position part of the array above as for different functions, the parameter position might be different and hardcoding it to `1` here does not allow for that flexibility.

See these example sniffs which also uses this utility function:
* https://github.com/wimg/PHPCompatibility/blob/master/Sniffs/PHP/NewFunctionParametersSniff.php
* https://github.com/wimg/PHPCompatibility/blob/master/Sniffs/PHP/RemovedFunctionParametersSniff.php
* https://github.com/wimg/PHPCompatibility/blob/master/Sniffs/PHP/RequiredOptionalFunctionParametersSniff.php

The array for WP would look differently, but you can probably get some inspiration from the array as used there.